service/resource/vaultcrt/current.go

```diff
@@ -19,9 +19,14 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 	r.logger.Log("cluster", key.ClusterID(customObject), "debug", "looking for the secret in the Kubernetes API")
 
+	ns := key.SecretNamespace(customObject)
+	if ns == "" {
+		ns = r.namespace
+	}
```

Comment:
Since this is only used within the scope block below I would like to put this code inside the block.

Reply:
yes.